src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -131,7 +132,7 @@ public EnvVars getEnvironment(TaskListener listener) throws IOException, Interru
      * @return user's name who triggered the promotion, or 'anonymous'
      */
     public String getUserName(){
-    	UserCause userClause=getCause(UserCause.class);
+        UserIdCause userClause=getCause(UserIdCause.class);
```

Comment:
UserCause and UserIdCause are different subclasses of Cause. When you change the <code>getCause()</code>'s class requirement, you may start getting nulls for causes, which were valid before. If an old plugin uses a deprecated UserCause, the plugin will miss it.

A support of UserIdCauses is definitely required, but I would recommend to use a UserCause as a fallback if the first <code>getCause(UserIdCause.class)</code> returns null. It should be in Jenkins core, but unfortunately is has not been taken into the account.